src/travel-details-map-screen/TravelDetailsMapScreenComponent.tsx

```diff
@@ -159,71 +175,108 @@ const styles = StyleSheet.create({
 });
 
 type VehicleIconProps = {
-  coordinates: Coordinates;
+  vehicle: VehicleWithPosition;
   mode?: AnyMode;
   subMode?: AnySubMode;
   setShouldTrack: React.Dispatch<React.SetStateAction<boolean>>;
+  subscriptionState: SubscriptionState;
+  zoomLevel: number;
 };
 
 const LiveVehicle = ({
-  coordinates,
+  vehicle,
   setShouldTrack,
   mode,
   subMode,
+  subscriptionState,
+  zoomLevel,
 }: VehicleIconProps) => {
+  const {theme} = useTheme();
+
+  const isError =
+    subscriptionState === 'CLOSING' || subscriptionState === 'CLOSED';
+  const isLoading =
+    subscriptionState === 'CONNECTING' || subscriptionState === 'NOT_STARTED';
+  const [isStale, setIsStale] = useState(false);
+
+  useInterval(
+    () => {
+      const secondsSinceUpdate = secondsBetween(
+        vehicle.lastUpdated,
+        new Date(),
+      );
+      setIsStale(STALE_DATA_THRESHOLD_IN_SECONDS < secondsSinceUpdate);
+    },
+    1000,
+    [vehicle.lastUpdated],
+    false,
+    true,
+  );
+
   const circleColor = useTransportationColor(mode, subMode);
-  const iconName = vehicleIconName(mode);
+  const textColor = useTransportationColor(mode, subMode, 'text');
+  const svg = getTransportModeSvg(mode, subMode);
+
+  const circleStyle = ((): CircleLayerStyleProps => {
+    if (isError)
+      return {
+        circleColor:
+          theme.interactive.interactive_destructive.disabled.background,
+        circleRadius: 20,
+        circleStrokeColor:
+          theme.interactive.interactive_destructive.default.background,
+        circleStrokeWidth: 2,
+      };
+    if (isStale)
+      return {
+        circleColor: theme.interactive.interactive_1.disabled.background,
+        circleRadius: 22,
+        circleStrokeWidth: 0,
+      };
+    return {
+      circleColor,
+      circleRadius: 22,
+      circleStrokeWidth: 0,
+    };
+  })();
 
-  if (!coordinates) return null;
+  const LiveVehicleIcon = (): JSX.Element => {
+    if (isLoading) return <ActivityIndicator color={textColor} />;
+    if (isError)
+      return (
+        <ThemeIcon
+          svg={svg}
+          fill={theme.interactive.interactive_destructive.default.background}
+        />
+      );
+    if (isStale)
+      return (
+        <ThemeIcon
+          svg={svg}
+          fill={theme.interactive.interactive_1.disabled.text}
+        />
+      );
+    return <ThemeIcon svg={svg} fill={textColor} />;
+  };
```

Comment:
I think it is preferable to drag out this LiveVehicleIcon into its own component. When you recreate the component function for each render you miss out on some of the react optimisation features which rerenders the component only on changes etc.